File: src/main/java/pl/allegro/experiments/chi/chiserver/infrastructure/interactions/InMemoryInteractionRepository.java
package pl.allegro.experiments.chi.chiserver.infrastructure.interactions;

import pl.allegro.experiments.chi.chiserver.domain.interactions.Interaction;
import pl.allegro.experiments.chi.chiserver.domain.interactions.InteractionRepository;

import java.util.ArrayList;
import java.util.List;

public class InMemoryInteractionRepository implements InteractionRepository {
    private final List<Interaction> interactions = new ArrayList<>();

    public void save(Interaction interaction) {
        interactions.add(interaction);
    }

    public boolean interactionSaved(Interaction interaction) {
        return interactions.contains(interaction);
    }
}

File: src/main/kotlin/pl/allegro/experiments/chi/chiserver/infrastructure/interactions/InMemoryInteractionRepository.kt
package pl.allegro.experiments.chi.chiserver.infrastructure.interactions

import pl.allegro.experiments.chi.chiserver.domain.interactions.Interaction
import pl.allegro.experiments.chi.chiserver.domain.interactions.InteractionRepository
import kotlin.collections.ArrayList

class InMemoryInteractionRepository : InteractionRepository {
    private val interactions: MutableList<Interaction> = ArrayList()

    override fun save(interaction: Interaction) {
        interactions.add(interaction)
    }

    fun interactionSaved(interaction: Interaction): Boolean {
        return interactions.contains(interaction)
    }
}